source/symbols/Type.cpp

```diff
@@ -481,24 +486,55 @@ bool Type::isAssignmentCompatible(const Type& rhs) const {
     return false;
 }
 
-bool Type::isCastCompatible(const Type& rhs) const {
+// Return 2 for bit-stream casting, 1 for valid explicit cast, 0 for invalid type cast
```

Comment:
Rather than using an int return value here, which isn't self documenting, I think it would make more sense to nest this function in a higher-level one called something like isBitstreamCastable(). Similar to how isCastCompatible calls into isAssignmentCompatible which calls into isEquivalent, isBitstreamCastable would call into isCastCompatible first.

Reply:
How about making "isBitstreamCastable" and "isCastCompatible" two independent functions without wrapping each other? The purpose is to distinguish them and if "isBitstreamCastable" calls "isCastCompatible", the return value of "isBitstreamCastable" cannot be bool and must have 3 values, which still needs extra documenting. There are two places calling "isCastCompatible" now. With two independent functions,
```c++
            type.isCastCompatible(*rt) ? diag::NoImplicitConversion : diag::BadAssignment;
```
can become
```c++
            type.isCastCompatible(*rt) || type.isBitstreamCapaible(*rt)? diag::NoImplicitConversion : diag::BadAssignment;
```
and
```c++
    auto castKind = type.isCastCompatible(*operand.type);
    if (!castKind) {
        auto& diag = context.addDiag(diag::BadConversion, syntax.apostrophe.location());
        ...
    if (castKind > 1)

```
can become something like
```c++
    bool isBitstreamCast = false;
    bool isCastCompatible =  type.isCastCompatible(*operand.type);
    if (!isCastCompatible)
        isBitstreamCast = type.isBitstreamCastable(*operand.type);
    if (!isCastCompatible && !isBitstreamCast) {
       auto& diag = context.addDiag(diag::BadConversion, syntax.apostrophe.location());
        ...
    if (isBitstreamCast)
```

Reply:
Yeah, that works too.